crates/topos-p2p/src/behaviour/gossip.rs

```diff
@@ -10,37 +10,35 @@ use libp2p::{
     identity::Keypair,
     swarm::{NetworkBehaviour, THandlerInEvent, ToSwarm},
 };
-use serde::{Deserialize, Serialize};
-use topos_metrics::{
-    P2P_DUPLICATE_MESSAGE_ID_RECEIVED_TOTAL, P2P_GOSSIP_BATCH_SIZE,
-    P2P_MESSAGE_SERIALIZE_FAILURE_TOTAL,
-};
+use prost::Message as ProstMessage;
+use topos_api::grpc::tce::v1::{Batch, DoubleEchoRequest};
+use topos_metrics::{P2P_DUPLICATE_MESSAGE_ID_RECEIVED_TOTAL, P2P_GOSSIP_BATCH_SIZE};
 use tracing::{debug, error};
 
 use crate::{constants, event::ComposedEvent, TOPOS_ECHO, TOPOS_GOSSIP, TOPOS_READY};
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub(crate) struct Batch {
-    pub(crate) data: Vec<Vec<u8>>,
-}
-
 pub struct Behaviour {
     batch_size: usize,
     gossipsub: gossipsub::Behaviour,
-    echo_queue: VecDeque<Vec<u8>>,
-    ready_queue: VecDeque<Vec<u8>>,
+    echo_queue: VecDeque<DoubleEchoRequest>,
+    ready_queue: VecDeque<DoubleEchoRequest>,
```

Comment:
Do you think that we can remove those two and switch to a `HashMap<&'static str, VecDeque<Vec<u8>>>`  so we can remove the code duplication in publish/poll methods and also remove the link between the Gossip behaviour and the double echo?